### PR TITLE
DataViews: Add story

### PR DESCRIPTION
**Pull request by ntsekouras**
## What?
This PR adds a basic story for DataViews component to enable UI development on isolation. I discovered a couple of issues already that are being addressed here.

## Testing Instructions
1. `npm run storybook:dev` and test the component.

### CODE REVIEW

File: packages/dataviews/src/style.scss [hunk 1]
```diff
@@ -3,6 +3,7 @@
 	height: calc(100% - #{$grid-unit-40} * 2);
```
Comment (ntsekouras) [inline packages/dataviews/src/style.scss:3]: There is [an issue](https://github.com/storybookjs/storybook/issues/8112) with storybook iframe's height that results in `sticky` headers not to work properly. I'm not sure if this should be treated as storybook issue or it indicates that we should handle somehow such a use case. --cc @ciampo

Comment (youknowriad) [reply packages/dataviews/src/style.scss:3]: How can we handle it in a way that works everywhere (regardless of how dataviews is used)

**Review by youknowriad — APPROVED** [review]
This is a very cool PR. 

It shows another issue for me. I believe we should have a way to disable `useAsyncList` by default and only enable it on demand. For this static performant data views, there's no need to enable it for instance.

File: packages/dataviews/src/style.scss (hunk 1 above)
Comment (ciampo) [reply packages/dataviews/src/style.scss:3]: You can test the story outside of the Storybook canvas iframe ([example](https://wordpress.github.io/gutenberg/iframe.html?args=&id=dataviews-experimental-dataviews--default&viewMode=story)) to understand if the iframe could be the issue here

**Comment by oandregal** [comment]
https://github.com/WordPress/gutenberg/pull/66727 fixes the order of styles, by loading component before dataviews' styles. The issue was raised at https://github.com/WordPress/gutenberg/pull/56760#issuecomment-2447233344

**Pull request merged**